eth/fetcher/tx_fetcher.go

```diff
@@ -300,10 +311,15 @@ func (f *TxFetcher) Enqueue(peer string, txs []*types.Transaction, direct bool)
 			// Avoid re-request this transaction when we receive another
 			// announcement.
 			if errors.Is(err, txpool.ErrUnderpriced) || errors.Is(err, txpool.ErrReplaceUnderpriced) {
-				for f.underpriced.Cardinality() >= maxTxUnderpricedSetSize {
-					f.underpriced.Pop()
+				// If the set is to big, delete a pseudorandom element
+				for hash := range f.underpriced {
```

Comment:
I would prefer to delete the oldest element if possible without introducing too much complexity. Although I know the requirement is not that strict.

Reply:
Yep, switched to lru now